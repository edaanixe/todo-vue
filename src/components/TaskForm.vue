<template>
    <div class="add-items d-flex">
        <input type="text" class="form-control todo-list-input" placeholder="What do you need to do today?" v-model="taskTitle" />
        <button class="add btn btn-primary font-weight-bold todo-list-add-btn" @click="$emit('add-task')">
            Add
        </button>
    </div>
</template>