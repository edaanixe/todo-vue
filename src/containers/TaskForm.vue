<template>
    <div class="add-items d-flex">
        <input type="text" class="form-control todo-list-input" placeholder="What do you need to do today?" v-model="title" />
        <button class="add btn btn-primary font-weight-bold todo-list-add-btn" @click="addTask()" :disabled="!title">
            Add
        </button>
    </div>
</template>

<script>
export default {
    data(){
        return {
            title: ''
        }
    },
    methods: {

        addTask(){
            this.$emit('add-task', this.title)
            this.title = ''
        }
        
    }
}
</script>