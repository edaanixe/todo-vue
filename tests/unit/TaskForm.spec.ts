import { shallowMount } from '@vue/test-utils'
import TaskForm from '@/components/TaskForm.vue'

describe('TaskForm', () => {
    it('should match element against snapshot', () => {
        const wrapper = shallowMount(TaskForm)
        expect(wrapper.element).toMatchSnapshot()
    })

    it('should trigger add-task event', async () => {
        const wrapper = shallowMount(TaskForm)

        await wrapper.find('.todo-list-add-btn').trigger('click')

        expect(wrapper.emitted('add-task')).toHaveLength(1)
    });
});