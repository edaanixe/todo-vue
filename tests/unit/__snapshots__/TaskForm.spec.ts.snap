// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm should match element against snapshot 1`] = `
<div
  class="add-items d-flex"
>
  <input
    class="form-control todo-list-input"
    placeholder="What do you need to do today?"
    type="text"
  />
   
  <button
    class="add btn btn-primary font-weight-bold todo-list-add-btn"
  >
    
        Add
    
  </button>
</div>
`;
